.gitlab/corona/pipeline.yml

```diff
@@ -56,42 +61,40 @@ allocate lc resources:
 
 # Build LBANN and establish the Spack environment for this pipeline.
 build and install:
-  extends: .corona common
+  extends:
+    - .corona common
+    - .lbann-base-vars
   stage: build
   artifacts:
     paths:
       - spack-*.txt
-      - spack-build-*/CMakeCache.txt
-      - spack-build-*/build.ninja
-      - spack-build-*/spack-*.txt
       - spack-ci-env-name.sh
+      - LBANN_*.cmake
+      - LBANN_*_setup_build_tools.sh
+      - LBANN_*_setup_module_path.sh
+      - builds/lbann_${SYSTEM_NAME}_${SPACK_ENV_BASE_NAME}-*${CI_CONCURRENT_ID}-*/*
```

Comment:
A note and a couple questions.

The note: this keeps the entire build and install directories. The artifacts were over 600MB for one job that I checked.

That's neither good nor bad, per se. I'd probably argue that keeping the build directory is unnecessary unless you're running the Catch2 tests in a downstream job. But if we're ok with forwarding the installed LBANN downstream as an artifact (vs the old approach where it was cached in some spack directory outside of CI's view), why is the path name so complicated? One of the advantages of this setup is that we get to choose the name, so why not pick something simple, like `build` and `install`? Then restoring everything is pretty straightforward -- `ml use $PWD/install/etc/modulefiles && ml lbann` and you're done (well, once you've found all the spack runtime crap).

Reply:
@benson31 We can simplify this if we want, but it means that builds will collide if the reuse the same repository for multiple builds or systems.

Reply:
We shouldn't be doing that in CI.